draft-ietf-httpbis-priority.md

```diff
@@ -108,101 +114,100 @@ The term control stream is used to describe the HTTP/2 stream with identifier
+porting the approach to protocols that do not provide global ordering. For
+example, it could not be supported in HTTP/3 {{HTTP3}} without changing the
+signals and their processing.
+
+Considering the problems with the deployment of RFC 7540 stream priority, and
+the difficulties in adapting it to HTTP/3, continuing to base prioritization on
+this mechanism risks increasing the complexity of systems. Multiple experiments
+from independent research have shown that simpler schemes can reach at least
+equivalent performance characteristics compared to the more complex RFC 7540
+setups seen in practice, at least for the web use case.
+
+## Disabling RFC 7540 Priorities {#disabling}
+
+The problems and insights set out above provided the motivation for deprecating
+RFC 7540 stream priority (see {{Section 5.3 of RFC7540}}).
+
+The SETTINGS_DEPRECATE_RFC7540_PRIORITIES setting is defined by this document in
+order to allow endpoints to explicitly opt out of using HTTP/2 priority signals
+(see {{Section 5.3.2 of HTTP2}}). Endpoints are expected to use an alternative,
+such as the scheme defined in this specification.
+
+The value of SETTINGS_DEPRECATE_RFC7540_PRIORITIES MUST be 0 or 1. Any value
+other than 0 or 1 MUST be treated as a connection error (see {{Section 5.4.1 of
+HTTP2}}) of type PROTOCOL_ERROR.
 
 Endpoints MUST send this SETTINGS parameter as part of the first SETTINGS frame.
-A sender MUST NOT change the SETTINGS_DEPRECATE_HTTP2_PRIORITIES parameter value
+A sender MUST NOT change the SETTINGS_DEPRECATE_RFC7540_PRIORITIES parameter value
 after the first SETTINGS frame. Detection of a change by a receiver MUST be
 treated as a connection error of type PROTOCOL_ERROR.
 
 Until the client receives the SETTINGS frame from the server, the client SHOULD
-send the signals of the HTTP/2 priority scheme  (see {{motivation}}) and the
-signals of this prioritization scheme (see {{header-field}} and
-{{h2-update-frame}}). When the client receives the first SETTINGS frame that
-contains the SETTINGS_DEPRECATE_HTTP2_PRIORITIES parameter with value of 1, it
-SHOULD stop sending the HTTP/2 priority signals. If the value was 0 or if the
-settings parameter was absent, it SHOULD stop sending PRIORITY_UPDATE frames
-({{h2-update-frame}}), but MAY continue sending the Priority header field
-({{header-field}}), as it is an end-to-end signal that might be useful to nodes
-behind the server that the client is directly connected to.
-
-The SETTINGS frame precedes any priority signal sent from a client in HTTP/2, so
-a server can determine if it should respect the HTTP/2 scheme before building
-state. A server that receives SETTINGS_DEPRECATE_HTTP2_PRIORITIES with value of
-1 MUST ignore HTTP/2 priority signals.
-
-Where both endpoints disable HTTP/2 priorities, the client is expected to send
-this scheme's priority signal. Handling of omitted signals is described in
+send both the HTTP/2 priority signals and the signals of this prioritization
```

Comment:
Should this be HTTP/2-bis or RFC 7540? IIUC, any specific details of these signals is removed from -bis, so it should be RFC 7540 to be able to use them in this way?

Reply:
```suggestion
send both the HTTP/2 priority signals and the signals of the employed alternative scheme
```

Above, you indicate any scheme can be used, with this being just one option. 

This does make guidance about PRIO_UPDATE frames/header field more difficult later, so maybe some other textual adjustment is needed earlier:

e.g., for the remainder of this text ,we assume the use of the scheme defined in this document as the alternate

Or, move the following text from below up and adjust a bit 
> Where both endpoints disable RFC 7540 stream priority, the client is expected to
send this scheme's priority signal

Reply:
I made https://github.com/httpwg/http-extensions/issues/1679 because the question is tangential to this PR.

Reply:
I would have gone for MAY here.

Reply:
Can we please continue the discussion on the issue?

PS: the requirement didn't change in this PR, so the discussion of this point is revisiting a design decision, so let's make sure its surfaced to folks.